volatility3/framework/symbols/linux/__init__.py

```diff
@@ -266,4 +280,13 @@ def walk_internal_list(cls, vmlinux, struct_name, list_member, list_start):
         while list_start:
             list_struct = vmlinux.object(object_type = struct_name, offset = list_start.vol.offset)
             yield list_struct
-            list_start = getattr(list_struct, list_member)
\ No newline at end of file
+            list_start = getattr(list_struct, list_member)
+
+    @classmethod
+    def container_of(cls, addr, type_name, member_name, vmlinux):
```

Comment:
Typing information would be useful here, particularly that the return is `Optional` so people can tell that it may just return None rather than throwing an exception on failure.

Reply:
This feels as though maybe it should be a method on an object (something like `get_container` or similar?).  This could use a docstring too.  I think we do something similar for list structures (doing maths on the relative offset), if that's the case, it might be worth drawing this out as something more general.  @atcuno care to comment?

Reply:
Docstring and typing info added. 
Regarding the method name, I get your point but in this case I think it would be better to keep the function name as is. That's a quite common macro in the Linux kernel and I think it's more natural to read, write/translate or compare code using the same name. Anyway, no problem if you decide `get_container` is a better name.

Reply:
container_of is the name of the corresponding macro from the Linux kernel so I would prefer to keep it this way

Reply:
Cool, that makes sense @atcuno.  Thanks @gcmoreira 5:)